server/src/test/java/org/elasticsearch/discovery/AbstractDisruptionTestCase.java

```diff
@@ -154,20 +157,11 @@ protected void beforeIndexDeletion() throws Exception {
         return Arrays.asList(MockTransportService.TestPlugin.class);
     }
 
-    void configureCluster(
-            int numberOfNodes,
-            @Nullable int[] unicastHostsOrdinals,
-            int minimumMasterNode
-    ) throws ExecutionException, InterruptedException {
-        configureCluster(DEFAULT_SETTINGS, numberOfNodes, unicastHostsOrdinals, minimumMasterNode);
+    void configureCluster(int numberOfNodes, int minimumMasterNode) {
+        configureCluster(DEFAULT_SETTINGS, numberOfNodes, minimumMasterNode);
     }
 
-    void configureCluster(
-            Settings settings,
-            int numberOfNodes,
-            @Nullable int[] unicastHostsOrdinals,
-            int minimumMasterNode
-    ) throws ExecutionException, InterruptedException {
+    void configureCluster(Settings settings, int numberOfNodes, int minimumMasterNode) {
         if (minimumMasterNode < 0) {
             minimumMasterNode = numberOfNodes / 2 + 1;
```

Comment:
As far as I can tell, we allow tests to set `minimumMasterNode` to the "wrong" value only to allow dynamically adding nodes after the cluster has started. If so, it'd be better to allow `ESIntegTestCase` to manage this number for us.

Reply:
@DaveCTurner Aren't we only ever passing `-1` here from `org.elasticsearch.discovery.AbstractDisruptionTestCase#startCluster(int, int)` when we start the cluster? It seems like this is just a convenience method for not having to set a value by hand here?

Reply:
Sorry, let me clarify. I do not think this method should receive an explicit `minimumMasterNode` value at all, whether `-1` or otherwise. In almost all cases, `ESIntegTestCase` now does a better job of automatically managing this setting than is done here. It was, however, possible that some of the tests were using this feature to set this value to something other than `numberOfNodes / 2 + 1`, for instance:

https://github.com/elastic/elasticsearch/blob/d7ef985a79ee18a5a567bfd50cc788ee6e0da457/server/src/test/java/org/elasticsearch/discovery/ClusterDisruptionIT.java#L366

https://github.com/elastic/elasticsearch/blob/d7ef985a79ee18a5a567bfd50cc788ee6e0da457/server/src/test/java/org/elasticsearch/discovery/SnapshotDisruptionIT.java#L62

https://github.com/elastic/elasticsearch/blob/8fe964de36e2f1dfa573257789b1e71c91aa8b7a/server/src/test/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java#L141

In the first two cases this is actually correct because `numberOfNodes` also includes some data-only nodes. The last case is kinda strange - I think it really can form two clusters, and I'm not sure what it's actually testing.

Reply:
@DaveCTurner given the short exchange in Slack, wanna keep this around for now maybe?

Reply:
Yes, let's do this in a follow-up. NB the action to take here is to make this a cluster with 1 master node and 1 data node, which'd then mean that we can fall back on `ESIntegTestCase`'s management of the cluster configuration.